### PR TITLE
Replace `got` and `axios` dependencies with `node-fetch`

### LINKED ISSUE #8553: Remove non-`fetch` http libraries
There are some remaining usages of `axios` and `got` (and more?) in the project. These should be removed, and replaced with `fetch`.

In backend packages, the `node-fetch` library should be used.

In frontend and isomorphic packages, the `cross-fetch` library should be used.

### PR DESCRIPTION
**Pull request by cmpadden**
Closes #8553.

## Hey, I just made a Pull Request!

This pull request replaces the following dependencies with `node-fetch`:

- `got` from `@backstage/plugin-auth-backend`
- `axios` from `@backstage/plugin-rollbar-backend`

A unit test was introduced to verify the changes to `rollbar-backend`, and while a unit test was not created for the Microsoft provider, testing was done via the Node REPL. See the attached file that demonstrates that the request for image data via `got` and `fetch` assert to be true.

[got_vs_fetch.js.txt](https://github.com/backstage/backstage/files/7764773/got_vs_fetch.js.txt)

While attempting to write the unit test for `auth-backend` another discovery was made which I will outline in a comment on this PR.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by cmpadden** [comment]
While writing a unit test for the `getUserPhoto` method in the Microsoft provider, I noticed that the existing unit test actually queries the real Microsoft Graph API at the following URL:

```
'https://graph.microsoft.com/v1.0/me/photos/48x48/$value'
```

This was discovered when the _base64_ encoding of the photo was changing each time the unit test ran. I added a log statement to the aforementioned method to see the response:

```
console.log(Buffer.from(arrayBuffer).toString('utf-8'))
```

```
{"error":{"code":"InvalidAuthenticationToken","message":"CompactToken parsing failed with error code: 80049217","innerError":{"date":"2021-12-22T19:01:34","request-id":"db2dbb5e-8435-449e-9764-0826d1f9d36a","client-request-id":"db2dbb5e-8435-449e-9764-0826d1f9d36a"}}}
```

What are peoples thoughts? Should we use `msw` to intercept and mock the image response in this scenario?

**Comment by Rugvip** [comment]
@cmpadden yeah I think it's best to intercept and mock. No need to use a real image though since we're just base64 encoding anyway. Could just return a couple of known bytes and make sure they come out alright

File: plugins/auth-backend/src/providers/microsoft/provider.test.ts [hunk 1]
```diff
@@ -103,4 +128,45 @@ describe('createMicrosoftProvider', () => {
...
+      callbackUrl: 'mock',
+      // define resolver to return user `info` for photo validation
+      signInResolver: async (info, _) => {
+        return {
+          id: 'user.name',
+          token: 'token',
+          info: info,
+        };
+      },
+    });
```
Comment (cmpadden) [inline plugins/auth-backend/src/providers/microsoft/provider.test.ts:165]: @rugvip Added a mocked & intercepted image response!

To validate the photo on the identity, I specified a _signInResolve_ method that returns the user's info. This is a bit of a hack, but I wasn't sure how else to access the `fullProfile.photos` attribute. Lmk if there's something better. Thanks.

**Review by jhaals — APPROVED** [review]
Nice work!👌

**Review by Rugvip — APPROVED** [review]
Yep, looks good, thank you! 👍 🎉

**Pull request merged**